File: backend/db/repositories/messages.py
from uuid import UUID

from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from models.message import Message


async def save_message(db: AsyncSession, session_id: UUID, role: str, content: str) -> Message:
    message = Message(session_id=session_id, role=role, content=content)
    db.add(message)
    await db.commit()
    await db.refresh(message)
    return message


async def get_history(db: AsyncSession, session_id: UUID) -> list[Message]:
    result = await db.execute(
        select(Message)
        .where(Message.session_id == session_id)
        .order_by(Message.created_at)
    )
    return result.scalars().all()

File: node-server/src/db/repositories/messages.ts
import { asc, eq } from "drizzle-orm";
import type { DbExecutor } from "../client.js";
import { messages, type Message } from "../schema.js";

export async function saveMessage(
  db: DbExecutor,
  sessionId: string,
  role: string,
  content: string,
): Promise<Message> {
  const [row] = await db.insert(messages).values({ sessionId, role, content }).returning();
  return row!;
}

export async function getHistory(db: DbExecutor, sessionId: string): Promise<Message[]> {
  return db
    .select()
    .from(messages)
    .where(eq(messages.sessionId, sessionId))
    .orderBy(asc(messages.createdAt));
}
